examples/policies/soul-bound-policy/soul-bound-policy-v1.pact

```diff
@@ -0,0 +1,120 @@
+(namespace (read-msg 'ns))
+
+(module soul-bound-policy-v1 GOVERNANCE
+
+  (defconst ADMIN-KS:string "marmalade-v2.marmalade-contract-admin")
+
+  (defcap GOVERNANCE ()
+    (enforce-guard ADMIN-KS))
+
+  (implements kip.token-policy-v2)
+  (use kip.token-policy-v2 [token-info])
+  (use marmalade-v2.policy-manager)
+  (use marmalade-v2.guard-policy-v1 [MINT-GUARD-MSG-KEY BURN-GUARD-MSG-KEY GUARD_SUCCESS])
+
+  (defschema mint-record 
+    account:string
+  )
+
+  (deftable records:{mint-record})
+
+  (defun has-guard-policy:bool (policies)
+    (> (length (filter (lambda (policy) (= policy marmalade-v2.guard-policy-v1)) policies)) 0))
+
+  (defun enforce-init:bool
+    ( token:object{token-info}
+    )
+    @doc "The function is run at `create-token` step of marmalade-v2.ledger."
+
+    (require-capability (INIT-CALL (at "id" token) (at "precision" token) (at "uri" token) soul-bound-policy-v1))
+
+    (enforce (= (at 'precision token) 0) "Precision must be 0 for soul-bound tokens")
+
+    (let* (
+        (policies (at 'policies token))
+
+        (mint-guard (try GUARD_SUCCESS (read-msg MINT-GUARD-MSG-KEY) ))
```

Comment:
`try` statement looks duplicative, as we'd like the function to fail when the guard keys are not available. 

We should also review the logic, if we'd like to check if we're going to allow open guards, because there are different ways to implement similar guards to `GUARD_SUCCESS`. 
We could only support `keysets` in this case, or specific principal'd keysets, with functions like `is-princpal`, and check to support only `k:` and `w:` accounts. 

```
        (policies:[module{kip.token-policy-v2}] (at 'policies token))

        (mint-guard:guard (read-msg MINT-GUARD-MSG-KEY) )
        (burn-guard:guard (read-msg BURN-GUARD-MSG-KEY) )
      )
```

Reply:
I removed the `try` logic and made it only read the msgs so it fails if you don't pass any, but the last part of your comment I do not understand fully. `MINT-GUARD-MSG-KEY` is a keyset so I can't pass it into `is-principal` as it expects a string, so there is no easy way to test it

Reply:
You're right, if we wanted to implement it a workaround would be something like 

`(create-principal  (read-keyset MINT-GUARD-MSG-KEY))` and checking its first index. 

But it was a suggested idea, and a question if we'd want to filter certain types of keysets.